src/pages/ProductList/index.js

```diff
@@ -1,52 +1,154 @@
-import { useState, useCallback } from "react";
+import { useState, useCallback, useMemo, useEffect } from "react";
 import { Products, Loader, SidebarItem, Pagination } from "../../components";
-import { useProducts } from "../../hooks";
+import { useQuery, useRequests } from "../../hooks";
 import classNames from "classnames";
-import categoriesData from "../../mocks/en-us/product-categories.json";
-import "./ProductList.scss";
+import { Button } from "../../styled";
+import {
+  WrapperProductList,
+  Sidebar,
+  SidebarTitle,
+  ClearFilters,
+  Content,
+} from "./styled";
+
+let PageSize = 12;
 
 function ProductList() {
-  const [filters, setFilters] = useState([]);
-  const [products, isLoading] = useProducts(filters);
+  let query = useQuery();
+  let categorySlug = query.get("category");
+
+  const requests = useMemo(
+    () => [
+      {
+        name: "productCategories",
+        params: {
+          q: '[[at(document.type, "category")]]',
+          pageSize: 30,
+        },
+      },
+      {
+        name: "productsList",
+        params: {
+          q: '[[at(document.type, "product")]]',
+          pageSize: 100,
+        },
+      },
```

Comment:
Consider moving these queries to your constants file as well. I don't think these would change. Perhaps create an entry for each request like:

```javascript
export const REQUEST = {
   PRODUC_LIST = ...
 };
```

And then you can use them like `REQUEST.PRODUCT_LIST` in here